infrastructure/src/main/java/org/corfudb/infrastructure/management/FailureDetector.java

```diff
@@ -158,10 +126,15 @@ private PollReport pollRound(
             PollReport currReport = pollIteration(allServers, router, epoch, sequencerMetrics, layout);
             reports.add(currReport);
 
-            long pollInterval = modifyIterationTimeouts(router, currReport);
+            Duration pollInterval = networkStretcher.modifyIterationTimeouts(currReport.getFailedNodes());
+            if (currReport.getFailedNodes().isEmpty()) {
```

Comment:
In case of a failure you need to tune router timeouts not otherwise.

Reply:
fixed